### PR TITLE
Move `runtime/{init_error.go,retriable_error.go}` into separate shared package

### PR DESCRIPTION
**Pull request by JoshVanL**
As part of a larger effort in breaking up runtime.go to be more modular so to eventually implement features like [hot reloading](https://github.com/dapr/dapr/issues/1172).

Moves the custom runtime init and retriable error types into a shared `runtime/errors` package.

### CODE REVIEW

File: pkg/runtime/errors/init.go [hunk 1]
```diff
@@ -43,8 +43,8 @@ func (e *InitError) Unwrap() error {
 	return e.err
 }
 
-// NewInitError returns an InitError wrapping an existing context error.
-func NewInitError(kind InitErrorKind, entity string, err error) *InitError {
+// NewInit returns an InitError wrapping an existing context error.
+func NewInit(kind InitErrorKind, entity string, err error) *InitError {
```
Comment (daixiang0) [inline pkg/runtime/errors/init.go:47]: I would like the old one since it is more clear, why change it?

File: pkg/runtime/errors/retriable.go [hunk 2]
```diff
@@ -32,8 +32,8 @@ func (e *RetriableError) Unwrap() error {
 	return e.err
 }
 
-// NewRetriableError returns a RetriableError wrapping an existing context error.
-func NewRetriableError(err error) *RetriableError {
+// NewRetriable returns a RetriableError wrapping an existing context error.
+func NewRetriable(err error) *RetriableError {
```
Comment (daixiang0) [inline pkg/runtime/errors/retriable.go:36]: Same as above.

File: pkg/runtime/errors/init.go (hunk 1 above)
Comment (JoshVanL) [reply pkg/runtime/errors/init.go:47]: This causes a stutter in the package function call:

```go
errors.NewInit
```

vs


```go
errors.NewInitError
```

**Review by ItalyPaleAle — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**